framework/src/meshgenerators/CartesianMeshGenerator.C

```diff
@@ -0,0 +1,368 @@
+      _subdomain_id = std::vector<unsigned int>(_dx.size() * _dy.size(), 0);
+    else
+      _subdomain_id = std::vector<unsigned int>(_dx.size(), 0);
+  }
+
+  // do dimension checks and expand block IDs for all sub-grids
+  switch (_dim)
+  {
+    case 1:
+    {
+      _nx = 0;
+      for (unsigned int i = 0; i < _dx.size(); ++i)
+        _nx += _ix[i];
+      _ny = 1;
+      _nz = 1;
+
+      std::vector<unsigned int> new_id;
+      for (unsigned int i = 0; i < _dx.size(); ++i)
+        for (unsigned int ii = 0; ii < _ix[i]; ++ii)
+          new_id.push_back(_subdomain_id[i]);
+
+      _subdomain_id = new_id;
+
+      if (isParamValid("dy"))
+        mooseWarning("dy provided for 1D");
+      if (isParamValid("iy"))
+        mooseWarning("iy provided for 1D");
+      if (isParamValid("dz"))
+        mooseWarning("dz provided for 1D");
+      if (isParamValid("iz"))
+        mooseWarning("iz provided for 1D");
+      break;
+    }
+    case 2:
+    {
+      _nx = 0;
+      for (unsigned int i = 0; i < _dx.size(); ++i)
+        _nx += _ix[i];
+      _ny = 0;
+      for (unsigned int i = 0; i < _dy.size(); ++i)
+        _ny += _iy[i];
+      _nz = 1;
+
+      std::vector<unsigned int> new_id;
+      for (unsigned int j = 0; j < _dy.size(); ++j)
+        for (unsigned int jj = 0; jj < _iy[j]; ++jj)
+          for (unsigned int i = 0; i < _dx.size(); ++i)
+            for (unsigned int ii = 0; ii < _ix[i]; ++ii)
+              new_id.push_back(_subdomain_id[j * _dx.size() + i]);
+
+      _subdomain_id = new_id;
+
+      if (!isParamValid("dy"))
+        mooseError("dy is not provided for 2D");
+      if (isParamValid("dz"))
+        mooseWarning("dz provided for 2D");
+      if (isParamValid("iz"))
+        mooseWarning("iz provided for 2D");
+      break;
+    }
+    case 3:
+    {
+      _nx = 0;
+      for (unsigned int i = 0; i < _dx.size(); ++i)
+        _nx += _ix[i];
+      _ny = 0;
+      for (unsigned int i = 0; i < _dy.size(); ++i)
+        _ny += _iy[i];
+      _nz = 0;
+      for (unsigned int i = 0; i < _dz.size(); ++i)
+        _nz += _iz[i];
+
+      std::vector<unsigned int> new_id;
+      for (unsigned int k = 0; k < _dz.size(); ++k)
+        for (unsigned int kk = 0; kk < _iz[k]; ++kk)
+          for (unsigned int j = 0; j < _dy.size(); ++j)
+            for (unsigned int jj = 0; jj < _iy[j]; ++jj)
+              for (unsigned int i = 0; i < _dx.size(); ++i)
+                for (unsigned int ii = 0; ii < _ix[i]; ++ii)
+                  new_id.push_back(_subdomain_id[k * _dx.size() * _dy.size() + j * _dx.size() + i]);
+
+      _subdomain_id = new_id;
+
+      if (!isParamValid("dy"))
+        mooseError("dy is not provided for 3D");
+      if (!isParamValid("dz"))
+        mooseError("dz is not provided for 3D");
+      break;
+    }
+  }
+}
+
+std::unique_ptr<MeshBase>
+CartesianMeshGenerator::generate()
+{
+  std::unique_ptr<ReplicatedMesh> mesh = libmesh_make_unique<ReplicatedMesh>(comm(), 2);
```

Comment:
This is where you are missing your include. This will change _very_ soon. I'm hoping to get my PR in today, but it's OK if this goes in first.

https://github.com/idaholab/moose/pull/12782/files#diff-fff72341d9af12d7f2d271967d56f915R107